### DISCUSSION TITLE
gene conversion rate f in ms vs msprime

### DISCUSSION THREAD

**Discussion by mshpak76** [tskit-dev/msprime#2161] [Q&A]
I am trying to implement an msprime script in ms to make sure that I am generating equivalent results. The one parameter that I'm struggling with converting is the gene conversion rate, specifically what to use for the f = g/r parameter in ms. In msprime, I have a per-site recombination rate of 1.14e-8 and a gene conversion rate of 5.9e-8 per site in each generation. In ms, f = g/r, where r is the per generation probability of crossing over between adjacent sites. I'm not sure how to convert this 5.9e-8 to the equivalent f = g/r. I assume r = 7.04e-8, the sum of conversion and recombination rates, but what is g? Is f simply the ratio 5.9/1.14 = 0.838 in this case, or is there some other scaling parameter that I'm missing?

Additionally, the scaling for the gene conversion rate isn't the same as the recombination rate in ms associated with the -r parameter, which isn't per site but rather per region/sequence length.

If somebody could clarify how to parameterize f wrt to the msprime equivalent rates, that would be very helpful.

**Comment by jeromekelleher** [comment]
Hi @mshpak76 :wave:

The code for converting from ms parameters to msprime equivalents is [here](https://github.com/tskit-dev/msprime/blob/c209ae52c10a48b53eecd3df260bd807bb18828d/msprime/cli.py#L424), hopefully that's help to answer your questions. Sorry I don't have time to go into more detail.

Perhaps this is something we should add to the [Switching from other simulators](https://tskit.dev/msprime/docs/stable/switch_from_other_simulators.html) section in the docs?

**Reply by fbaumdicker** [reply]
Hi @mshpak76 ,
if this is still relevant here is a short explanation:
Neither ms nor msprime use the sum of conversion and recombination rates as an explicit parameter.
In ms r corresponds to the total recombination rate, so the recombination rate per site times your number of loci -1.
In ms g corresponds to the total gene conversion initiation rate, so again the rate per site times your number of loci -1.
However, g is not a direct input of ms. Instead, ms only uses the ratio f to determine g.
Consequently, f = g/r is 5.9/1.14 = 5.1754  and not 5.9/7.04 = 0.838.

Note that this is only true for r > 0, otherwise for r = 0 f will be used as g.
